reana_client/utils.py

```diff
@@ -127,3 +130,104 @@ def _validate_reana_yaml(reana_yaml):
         logging.info('Invalid REANA specification: {error}'
                      .format(error=e.message))
         raise e
+
+
+def get_current_click_command(ctx):
+    """Return command (and subcommands) that were used to create given context.
+
+    For example, a context that was created by a command
+    '<executable name> code upload', would return 'code.upload'.
+
+    """
+    command = []
+    while ctx.parent:
+        command.append(ctx.info_name)
+        ctx = ctx.parent
+    return '.'.join(str(x) for x in reversed(command))
+
+
+def upload_files_to_server(user, organization, workflow, filenames, client):
+    """Upload files to REANA-Server.
+
+    Shared e.g. by `code upload` and `inputs upload`.g
+
+    :param user: User ID
+    :param organization: Organization ID
+    :param workflow: ID of that Workflow whose workspace should be
+    used to store the files.
+    :param filenames: Absolute filepath(s) of files to be uploaded.
+    :param client: Instance of Bravado Client.
+    """
+    if workflow:
+        logging.info('Workflow "{}" selected'.format(workflow))
+        if filenames:
+            for filename in filenames:
+                try:
+                    # Check if input is a folder and upload everything
+                    # including subdirectories.
+                    if os.path.isdir(filename):
+                        # TODO: Implement actual folder upload (with globbing)
```

Comment:
@tiborsimko should we move `TODO`s into issues and removing them from the code ?